numpy/core/src/multiarray/ctors.c

```diff
@@ -1113,30 +1146,7 @@ PyArray_NewFromDescr_int(
             data = npy_alloc_cache(nbytes);
         }
         if (data == NULL) {
-            static PyObject *exc_type = NULL;
-            
-            npy_cache_import(
-                "numpy.core._exceptions", "_ArrayMemoryError",
-                &exc_type);
-            if (exc_type == NULL) {
-                return NULL;
-            }
-            
-            PyObject *shape = PyArray_IntTupleFromIntp(fa->nd,fa->dimensions);
-            if (shape == NULL) {
-                return NULL;
-            }
-            
-            /* produce an error object */
-            PyObject *exc_value = PyTuple_Pack(2, shape, descr);
-            Py_DECREF(shape);
-            if (exc_value == NULL){
-                return NULL;
-            }
-            PyErr_SetObject(exc_type, exc_value);
-            Py_DECREF(exc_value);
-            return NULL;
-
+            return raise_memory_error(fa->nd, fa->dimensions, descr);
```

Comment:
```suggestion
            return raise_memory_error(fa->nd, fa->dimensions, (PyObject *)descr);
```
Travis/gcc is unhappy about the missing cast unfortunately. Otherwise looks perfect to me.